### ISSUE TITLE
Segfault loop after mastodon clsoes the stream

### ISSUE THREAD

**Issue by TheTincho** [kensanata/bitlbee-mastodon#35]
Hi, I am experiencing a problem similar to #7, but with version 1.4.1 of the plugin (debian stable). Could it be that this issue is not completely fixed?

In my case, bitlbee enters a segfault loop every time I reconnect, and I had to kill bitlbee completely and start it again to make it usable. I found that the start of the segfault loop is after getting a "Stream closed" error from Mastodon.

```
[17:17:35] <root> mastodon - Error: Stream closed (200 OK)
[17:17:35] <root> mastodon - Signing off..
[17:17:35] <root> mastodon - Logging in: Reconnecting in 5 seconds..
[17:17:40] <root> mastodon - Logging in: Login
[17:17:40] <root> mastodon - Logging in: Connecting
[17:17:40] <root> mastodon - Logging in: Verifying credentials
[17:17:40] <root> mastodon - Logging in: Getting home timeline
[17:17:40] <root> mastodon - Logging in: Logged in

[17:17:41] <*status> Error from Server [BitlBee crashed! (SIGSEGV received)]
[17:17:41] <*status> Disconnected from IRC. Reconnecting...
[17:17:43] <*status> Connected!
[17:17:46] <*status> Disconnected from IRC. Reconnecting...
[17:18:11] <*status> Connection Refused.  Reconnecting...
[17:18:42] <*status> Connection Refused.  Reconnecting...
[17:19:14] <*status> Connected!
[17:19:17] <*status> Error from Server [BitlBee crashed! (SIGSEGV received)]
[17:19:17] <*status> Disconnected from IRC. Reconnecting...
```

**Comment by kensanata** [comment]
Hm. Always a possibility, unfortunately. It works for me so I wonder whether you're doing anything suspicious? Filters? Hashtags?

**Comment by TheTincho** [comment]
Nope. Just getting the home timeline as a channel

**Comment by kensanata** [comment]
Hm. Can I create an account on the instance you're using? I feel like I'd have to reproduce it before I have a chance of doing anything.

**Comment by kensanata** [comment]
Hm. Another thought: do you have a Twitter account configured? Can you connect if you take your Twitter account offline? I just added a [Bugs](https://github.com/kensanata/bitlbee-mastodon#bugs) section to the README and linked to [a longer discussion elsewhere](https://alexschroeder.ch/software/Crashing_while_running_Twitter).

**Comment by TheTincho** [comment]
@kensanata sadly, my server is private, but I could give you access if we discuss this privately.

About the other question: I do have a twitter account configured. I can try doing that, but bear in mind that this bug is not always triggered: I was using the mastodon plugin for a few days before the segfaulting happened.

**Comment by kensanata** [comment]
Yeah, I guess I'll have to get myself a Twitter account just to test this weird interaction. :)
If you can setup your system such that the Mastodon account doesn't come online automatically and needs to be set online manually, and you get no more crashes, that would suggest that you're suffering from the same problem mentioned on that wiki page. As I'm currently travelling I won't be doing much about this anyway, so getting an account can wait.

**Comment by TheTincho** [comment]
I have been using both accounts for a few days, and I have not seen the issue reappear, so this seems plausible. Now, having said that, if I ever need to reauth to bitlbee, the problem starts again as I can't control which account logs in first: last time I had to uninstall the mastodon plugin just to be able to log in, disable mastodon, restart bitlbee with the plugin and try again..
